src/sql/workbench/services/notebook/browser/models/clientSession.ts

```diff
@@ -219,29 +219,31 @@ export class ClientSession implements IClientSession {
 	/**
 	 * Change the current kernel associated with the document.
 	 */
-	async changeKernel(options: nb.IKernelSpec, oldValue?: nb.IKernel): Promise<nb.IKernel> {
+	async changeKernel(options: nb.IKernelSpec, oldValue?: nb.IKernel): Promise<nb.IKernel | undefined> {
 		this._kernelChangeCompleted = new Deferred<void>();
 		this._isReady = false;
 		let oldKernel = oldValue ? oldValue : this.kernel;
 
 		let kernel = await this.doChangeKernel(options);
 		try {
-			await kernel.ready;
+			await kernel?.ready;
 		} catch (error) {
 			// Cleanup some state before re-throwing
-			this._isReady = kernel.isReady;
+			this._isReady = kernel ? kernel.isReady : false;
 			this._kernelChangeCompleted.resolve();
 			throw error;
 		}
-		let newKernel = this._session ? kernel : this._session.kernel;
-		this._isReady = kernel.isReady;
+		let newKernel = this._session ? this._session.kernel : kernel;
+		this._isReady = kernel ? kernel.isReady : false;
 		await this.updateCachedKernelSpec();
 		// Send resolution events to listeners
-		await this.notifyKernelChanged(oldKernel, newKernel);
+		if (newKernel) {
+			await this.notifyKernelChanged(newKernel, oldKernel);
+		}
 		return kernel;
 	}
 
-	private async notifyKernelChanged(oldKernel: nb.IKernel, newKernel: nb.IKernel): Promise<void> {
+	private async notifyKernelChanged(newKernel: nb.IKernel, oldKernel?: nb.IKernel): Promise<void> {
```

Comment:
Since we're reordering these arguments here, you might want to double check other places where this gets used. Both arguments are kernels, so the reordering won't show up as a compilation error.